ntp-daemon/src/spawn/mod.rs

```diff
@@ -0,0 +1,240 @@
+use std::{net::SocketAddr, sync::atomic::AtomicU64};
+
+use ntp_proto::PeerNtsData;
+use tokio::sync::mpsc;
+
+use crate::config::NormalizedAddress;
+
+#[cfg(test)]
+pub mod dummy;
+pub mod nts;
+pub mod pool;
+pub mod standard;
+
+/// Unique identifier for a spawner
+/// This is used to identify which spawner was used to create a peer
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub struct SpawnerId(u64);
+
+impl SpawnerId {
+    pub fn new() -> SpawnerId {
+        static COUNTER: AtomicU64 = AtomicU64::new(1);
+        SpawnerId(COUNTER.fetch_add(1, std::sync::atomic::Ordering::Relaxed))
```

Comment:
this is fancy, but does it not cause problems when testing?

Reply:
Should not be a problem! This should generate unique ids in any random combination of threads, ordering does not matter as there is no reliance on the exact number being generated here, the only requirement is for it to be a unique number.